### PR TITLE
Retain order blu-ray segments

### PR DESCRIPTION
**Pull request by Bond-009**
This fixes the order, but playback and chapter image extraction still don't work properly for me

### CODE REVIEW

File: MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs [hunk 1]
```diff
@@ -1137,13 +1137,11 @@ public IReadOnlyList<string> GetPrimaryPlaylistM2tsFiles(string path)
             var validPlaybackFiles = _blurayExaminer.GetDiscInfo(path).Files;
 
             // Get all files from the BDMV/STREAMING directory
-            var directoryFiles = _fileSystem.GetFiles(Path.Join(path, "BDMV", "STREAM"));
-
             // Only return playable local .m2ts files
-            return directoryFiles
-                .Where(f => validPlaybackFiles.Contains(f.Name, StringComparer.OrdinalIgnoreCase))
+            return validPlaybackFiles
+                .Select(f => _fileSystem.GetFileInfo(Path.Join(path, "BDMV", "STREAM", f)))
```
Comment (cvium) [inline MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:1142]: Won't this read every file individually?

Comment (Bond-009) [reply MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:1142]: One fstat per file, yes
Compared to one opendir and a readdir for each file in that directory

**Review by crobibero — APPROVED** [review]

**Review by Shadowghost — APPROVED** [review]

File: MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs (hunk 1 above)
Comment (cvium) [reply MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:1142]: Is that how EnumerateFiles works?

Comment (Bond-009) [reply MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:1142]: https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IO/Enumeration/FileSystemEnumerator.Unix.cs,180

**Review by cvium — APPROVED** [review]

**Pull request merged**